### PR TITLE
Closes #652 query datasets scope and scope num assertion

### LINKED ISSUE #652: Query datasets: SCOPE_NUM
### Idea
Add check on consistency of SCOPE and SCOPE_NUM (issue a warning at least)
See: https://github.com/Roche-GSK/admiral/pull/633#discussion_r746366314
Possibly use assert_one_to_one().
Check if there are any queries where SCOPE is missing but SCOPE_NUM is populated.

### Pseudo code

@kabis-ops - please review the description and add the remaining details above.

### PR DESCRIPTION
**Pull request by zdz2101**
Thank you for your Pull Request! We have developed this task checklist from the [Development Process Guide](https://pharmaverse.github.io/admiraldev/devel/articles/development_process.html) to help with the final steps of the process. Completing the below tasks helps to ensure our reviewers can maximize their time on your code as well as making sure the admiral codebase remains robust and consistent.   

Please check off each taskbox as an acknowledgment that you completed the task or check off that it is not relevant to your Pull Request. This checklist is part of the Github Action workflows and the Pull Request will not be merged into the `devel` branch until you have checked off each task.

- [x] Place Closes #<insert_issue_number> into the beginning of your Pull Request Title (Use Edit button in top-right if you need to update)
- [x] Code is formatted according to the [tidyverse style guide](https://style.tidyverse.org/). Run `styler::style_file()` to style R and Rmd files
- [x] Updated relevant unit tests or have written new unit tests, which should consider realistic data scenarios and edge cases, e.g. empty datasets, errors, boundary cases etc. - See [Unit Test Guide](https://pharmaverse.github.io/admiraldev/devel/articles/unit_test_guidance.html#tests-should-be-robust-to-cover-realistic-data-scenarios)
- [x] If you removed/replaced any function and/or function parameters, did you fully follow the [deprecation guidance](https://pharmaverse.github.io/admiraldev/devel/articles/programming_strategy.html#deprecation)?
- [x] Update to all relevant roxygen headers and examples, including keywords and families. Refer to the [categorization of functions](https://pharmaverse.github.io/admiraldev/devel/articles/programming_strategy.html#categorization-of-functions) to tag appropriate keyword/family.
- [x] Run `devtools::document()` so all `.Rd` files in the `man` folder and the `NAMESPACE` file in the project root are updated appropriately
- [x] Address any updates needed for vignettes and/or templates
- [x] Update `NEWS.md` if the changes pertain to a user-facing function (i.e. it has an `@export` tag) or documentation aimed at users (rather than developers)
- [x] Build admiral site `pkgdown::build_site()` and check that all affected examples are displayed correctly and that all new functions occur on the "[Reference](https://pharmaverse.github.io/admiral/reference/index.html)" page. 
- [x] Address or fix all lintr warnings and errors - `lintr::lint_package()`
- [x] Run `R CMD check` locally and address all errors and warnings - `devtools::check()`
- [x] Link the issue in the Development Section on the right hand side.
- [x] Address all merge conflicts and resolve appropriately 
- [x] Pat yourself on the back for a job well done! Much love to your accomplishment!

### CODE REVIEW

**Review by bms63 — COMMENTED** [review]
I think we should add a blurb in the news that this now throws a warning

**Review by bms63 — COMMENTED** [review]
Are the tests not following the preferred formatting?  Could you update or make a new issue for someone else to fix

File: tests/testthat/test-derive_vars_query.R [hunk 1]
```diff
@@ -150,7 +156,8 @@ test_that("Derive decides between TERM_NAME and TERM_ID based on the type of the
 })
 
 
-test_that("assert_valid_queries checks VAR_PREFIX values", {
+## Test 6: assert_valid_queries checks ----
+test_that("derive_vars_query Test 6: assert_valid_queries checks", {
```
Comment (bundfussr) [inline tests/testthat/test-derive_vars_query.R:160]: The test tests `assert_valid_queries()`. Thus the test description should start with "assert_valid_queries".

**Review by bundfussr — CHANGES_REQUESTED** [review]

File: R/derive_vars_query.R [hunk 2]
```diff
@@ -331,4 +331,17 @@ assert_valid_queries <- function(queries, queries_name) {
...
+    if (any((is.na(queries$QUERY_SCOPE) | queries$QUERY_SCOPE == "") &
+      !is.na(queries$QUERY_SCOPE_NUM))) {
+      abort(paste0(
+        "If a value for `QUERY_SCOPE_NUM` is specified",
+        "a corresponding  `QUERY_SCOPE` is expected",
+        " in `", queries_name, "`. ",
+        "They must both be NA/missing or both specified."
+      ))
+    }
+  }
```
[the remark this answers is not in the document]
Comment (zdz2101) [reply R/derive_vars_query.R:346]: I originally used `assert_one_to_one` for the two columns but it broke/gave warnings in the occds vignette, think I need clarification on exactly what the query structure should be as I'm not quite as familiar with it

image

[the remark this answers is not in the document]
Comment (bundfussr) [reply R/derive_vars_query.R:346]: I see. Looks like a mistake in the `queries` dataset. Could you fix it?

**Review by bundfussr — APPROVED** [review]

**Pull request merged**